content/en/docs/components/pipelines/v2/compile-a-pipeline.md

```diff
@@ -1,104 +1,75 @@
 weight = 5
 +++
 
-While pipelines and components are authored in Python, they can be compiled to intermediate representation (IR) YAML ([example][compiled-output-example]).
+You can compile your pipeline or component to intermediate representation (IR) YAML.  The IR YAML definition preserves a static representation of the pipeline or component. You can submit the YAML definition to the KFP backend for execution, or deserialize it using the KFP SDK for integration into another pipeline.
+([View an example on Github][compiled-output-example]).
 
-A YAML pipeline/component definition preserves a static representation of your pipeline/component. This YAML can be submitted to the KFP backend for execution or deserialized by the KFP SDK for integration into another pipeline.
+**Note:** Pipelines as well as components are authored in Python. A pipeline is a template representing a multistep workflow, whereas a component is a template representing a single step workflow.
 
-## Compiling
-First let's define a very simple pipeline:
+## Compile a pipeline
 
-```python
-from kfp import compiler
-from kfp import dsl
+Pipelines as well as components are workflow templates authored in Python. A pipeline represents a multistep workflow, whereas a component represents a single step workflow. You can compile a pipeline or a template into IR YAML using the `Compiler.compile` method. To do this, follow these steps:
 
-@dsl.component
-def addition_component(num1: int, num2: int) -> int:
-    return num1 + num2
+1.  Define a simple pipeline:
 
-@dsl.pipeline(name='addition-pipeline')
-def my_pipeline(a: int, b: int, c: int = 10):
-    add_task_1 = addition_component(num1=a, num2=b)
-    add_task_2 = addition_component(num1=add_task_1.output, num2=c)
-```
+    ```python
+    from kfp import compiler
+    from kfp import dsl
 
-Now we can compile the pipeline to the file `my_pipeline.yaml`:
+    @dsl.component
+    def addition_component(num1: int, num2: int) -> int:
+      return num1 + num2
 
-```python
-cmplr = compiler.Compiler()
-cmplr.compile(my_pipeline, package_path='my_pipeline.yaml')
-```
+    @dsl.pipeline(name='addition-pipeline')
+    def my_pipeline(a: int, b: int, c: int = 10):
+      add_task_1 = addition_component(num1=a, num2=b)
+      add_task_2 = addition_component(num1=add_task_1.output, num2=c)
+    ```
 
-Just as a pipeline is a template for a multi-step workflow, a component is a template for a single-step workflow. We can also compile the component `addition_component` directly:
+1.  Compile the pipeline to the file `my_pipeline.yaml`:
 
-```python
-cmplr.compile(addition_component, package_path='addition_component.yaml')
-```
+    ```python
+    cmplr = compiler.Compiler()
+    cmplr.compile(my_pipeline, package_path='my_pipeline.yaml')
+    ```
 
-The `Compiler.compile` method accepts a few optional additional parameters:
+1.  _Optional:_ Directly  compile the component `addition_component` to the file `addition_component.yaml`:
```

Comment:
I'm not sure this is "optional" toward the goal of compiling a pipeline, but a different goal altogether: compiling a component. Do you think we can find a middle ground between this structure and that provided in the original draft?

Reply:
Just read through the code. Removed the "optional" flag.